pytket/qir/conversion/api.py

```diff
@@ -83,3 +85,33 @@ def pytket_to_qir(
         return populated_module.module.ir()
     else:
         raise ValueError("unsupported return type")
+
+
+def pytket_to_qir_optimised(
+    circ: Circuit,
+    name: str = "Generated from input pytket circuit",
+    returntype: ReturnTypeQIR = ReturnTypeQIR.BINARY,
+) -> Union[str, bytes, None]:
+    """converts given pytket circuit to qir
+    :param circ: given circuit
+    :type circ: pytket circuit
+    :param name: name for the qir module created
+    :type name: str
+    :param returntype: format of the generated qir, defaut value is binary
+    :type returntype: ReturnTypeQIR
+    """
+
+    initial_result = str(pytket_to_qir(circ, name, ReturnTypeQIR.STRING))  # type: ignore
+
+    initial_result = initial_result.replace("entry_point", "EntryPoint")  # type: ignore
+    initial_result = initial_result.replace("num_required_qubits", "requiredQubits")  # type: ignore
+    result = initial_result.replace("num_required_results", "requiredResults")  # type: ignore
+
+    bitcode = pyqir.Module.from_ir(pyqir.Context(), result).bitcode  # type: ignore
```

Comment:
No need to compute this unless it's required.

Reply:
I wanted to have this in all cases to make sure that the replacements that we are doing still generated results that can be read by pyqir.